### PR TITLE
Add dust metrics to getblockstats

### PR DESCRIPTION
**Pull request by patricklodder**
Addition on top of #3297, to get more relevant stats.

adds:
- `maxoutamount` - highest value output
- `minoutamount` - lowest value output
- `dustouts` - number of outputs under our soft dust limit

Each of these ignores `OP_RETURN` and coinbase outputs

This makes some assumptions about `getblockstats` fixes. Draft until #3297 gets merged in any form.

### CODE REVIEW

File: src/rpc/blockchain.cpp [hunk 1]
```diff
@@ -1498,10 +1531,334 @@ UniValue reconsiderblock(const JSONRPCRequest& request)
...
+    if (size == 0) {
+        return 0;
+    }
+
+    std::sort(scores.begin(), scores.end());
+    if (size % 2 == 0) {
+        return (scores[size / 2 - 1] + scores[size / 2]) / 2;
+    } else {
+        return scores[size / 2];
+    }
```
Comment (chromatic) [inline src/rpc/blockchain.cpp:1547]: For my own clarification: if there's an even number of scores, there's no single median to choose, so average the two scores before and at the midpoint. If there's an odd number of scores, there's a median value?

File: src/validation.cpp [hunk 2]
```diff
@@ -1518,6 +1518,33 @@ bool CheckInputs(const CTransaction& tx, CValidationState &state, const CCoinsVi
...
+    CHashWriter hasher(SER_GETHASH, PROTOCOL_VERSION);
+    hasher << hashBlock;
+    hasher << blockundo;
+    if (hashChecksum != hasher.GetHash())
+        return error("%s: Checksum mismatch", __func__);
+
+    return true;
+}
+
 namespace {
```
Comment (chromatic) [inline src/validation.cpp:1548]: This looks the same as the version deleted from later in the file. Is that correct?

**Comment by patricklodder** [comment]
@chromatic: both comments are on code from #3297 - that needs to be merged first and this rebased on that.

File: src/validation.cpp (hunk 2 above)
Comment (patricklodder) [reply src/validation.cpp:1548]: Correct. It needs to be outside anon namespace to be usable outside. So this moves from internal validation function to an exported one through the change in validation.h below.

**Comment by patricklodder** [comment]
Rebased on top of 1.15.0-dev as of ba6480d3c

**Review by chromatic — APPROVED** [review]
ACK, all tests pass on x86-64 Linux.

The indentation is a little different in the new code (looks like Ruby style), but it doesn't introduce tabs where there weren't tabs, so it's not worth changing to me.

**Pull request merged**